From 7178784f0a94e2e6c668f587665fde41d405a23c Mon Sep 17 00:00:00 2001
From: Dmitry Kasatkin <d.kasatkin@samsung.com>
Date: Fri, 3 Oct 2014 14:40:20 +0300
Subject: [PATCH] ima: ignore empty and with whitespaces policy lines

Empty policy lines cause parsing failures which is, especially
for new users, hard to spot. This patch prevents it.

Changes in v2:
* strip leading blanks and tabs in rules to prevent parsing failures

Signed-off-by: Dmitry Kasatkin <d.kasatkin@samsung.com>
Signed-off-by: Mimi Zohar <zohar@linux.vnet.ibm.com>
---
 security/integrity/ima/ima_policy.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/security/integrity/ima/ima_policy.c b/security/integrity/ima/ima_policy.c
index bf232b98011e..d2c47d4df7b7 100644
--- a/security/integrity/ima/ima_policy.c
+++ b/security/integrity/ima/ima_policy.c
@@ -696,8 +696,9 @@ ssize_t ima_parse_add_rule(char *rule)
 
 	p = strsep(&rule, "\n");
 	len = strlen(p) + 1;
+	p += strspn(p, " \t");
 
-	if (*p == '#')
+	if (*p == '#' || *p == '\0')
 		return len;
 
 	entry = kzalloc(sizeof(*entry), GFP_KERNEL);
-- 
2.20.1 (Apple Git-117)

